src/bentoml/_internal/store.py

```diff
@@ -170,14 +170,26 @@ def register(self, tag: t.Union[str, Tag]):
         self._fs.makedirs(item_path)
         try:
             yield self._fs.getsyspath(item_path)
-        finally:
+        except Exception:
+            raise
+        else:
             # item generation is most likely successful, link latest path
-            if (
-                not self._fs.exists(_tag.latest_path())
-                or self.get(_tag).creation_time >= self.get(_tag.name).creation_time
-            ):
-                with self._fs.open(_tag.latest_path(), "w") as latest_file:
-                    latest_file.write(_tag.version)
+            self.update_latest(_tag)
+
+    def exists(self, tag: str | Tag) -> bool:
+        return self._fs.exists(Tag.from_taglike(tag).path())
+
+    def path_of(self, path: str) -> str:
+        return self._fs.getsyspath(path)
```

Comment:
Why are you putting these here and not using the `Bento`'s `path_of`?

Reply:
This is the store fs, not the bento. Bento and Model has its own path_of

Reply:
Right, but when do you think that this will be useful? I think we should enforce that people should use `get` and then use the Bento's FS.

Reply:
Do we want to expose the store API in the future? I think it is useful in general

Reply:
No, I don't think we should.